app/controllers/concerns/error_handler.rb

```diff
@@ -19,7 +19,12 @@ def self.included(clazz)
 
       rescue_from Errors::AuthError do |exception|
         log_exception(exception)
-        render json: { errors: [Types::ApplicationErrorType.root_level_from_application_error(exception)] }, status: :unauthorized
+        if exception.code == :not_found
+          status = :not_found
+        else
+          status = :unauthorized
+        end
+        render json: { errors: [Types::ApplicationErrorType.root_level_from_application_error(exception)] }, status: status
```

Comment:
Nice, #super minor, we could make it something like 👇 too
```suggestion
        http_status = exception.code == :not_found ? :not_found : :unauthorized
        render json: { errors: [Types::ApplicationErrorType.root_level_from_application_error(exception)] }, status: http_status
```